IQSS/dataverse
# Non-integer custom identifiers using stored procedure

This is my first contribution to Dataverse and first time I code Java, so please be understanding! 😄 

**What this PR does / why we need it**:

This PR replaces the option `sequentialNumber` for `:IdentifierGenerationStyle` with `storedProcGenerated`, and allows generating non-integer custom identifiers.

**Which issue(s) this PR closes**:

Closes #7548.

**Special notes for your reviewer**:

TODO:

- [x] Update [Documentation](https://guides.dataverse.org/en/latest/installation/config.html#identifiergenerationstyle) : which kind of example would you like to have in docs? The same as now?
- [x] Unit Tests? Not sure how to test this.

**Suggestions on how to test this**:

Tested on my local installation with the SQL script [createidentifier.txt](https://github.com/IQSS/dataverse/files/6732331/createidentifier.txt), which generates sequential identifiers from a base36 representation of current timestamp.

`[from L.A.:]` A very good test would be:
- Configure a 5.5 instance to use sequential identifiers (using the instructions from the 5.5 guide: execute the script `createsequence.sql`, and add one setting;)
- Confirm that you're getting sequential numeric identifiers; 
- Upgrade to this build; confirm that the instance is still assigning numeric identifiers. 

(so the goal is to confirm that Flyway successfully upgrades an existing old-style stored procedure)
`[/]`

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

No.

**Is there a release notes update needed for this change?**:

Probably yes.

**Additional documentation**:

A "flyaway script" was mentioned here https://github.com/IQSS/dataverse/issues/7548#issuecomment-771986937. Not sure where it should go and where it should be mentioned in documentation.
